### PR TITLE
add rtl support & refactor code

### PR DESCRIPTION
**Pull request by amit-mnnit**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/react-vanilla-components/src/components/tabs/HorizontalTab.tsx [hunk 1]
```diff
@@ -39,36 +37,59 @@ const HorizontalTab = (props: PROPS_PANEL) => {
...
-      </div>
+    <div
+      data-cmp-is="adaptiveFormTabs"
+      id={id}
+      className={`cmp-tabs ${appliedCssClassNames || ''}`}
+      data-placeholder-text="false"
+      data-cmp-visible={visible}
+      data-cmp-enabled={enabled}
+    >
+      <FieldWrapper
```
Comment (rismehta) [inline packages/react-vanilla-components/src/components/tabs/HorizontalTab.tsx:57]: This is not a field, code is not readable. Please rename FieldWrapper accordingly

File: packages/react-vanilla-components/src/components/common/FieldWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React, { useCallback, useState } from "react";
+import LabelContainer from "./LabelContainer";
+import Description from "./Description";
+import HelpContainer from "./HelpContainer";
```
Comment (rismehta) [inline packages/react-vanilla-components/src/components/common/FieldWrapper.tsx:4]: Exporting modules that are used exclusively within a single file can add unnecessary complexity and may make the code less maintainable. We should keep these functional components self contained

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: packages/react-vanilla-components/src/components/tabs/HorizontalTab.tsx (hunk 1 above)
Comment (amit-mnnit) [reply packages/react-vanilla-components/src/components/tabs/HorizontalTab.tsx]: changed

File: packages/react-vanilla-components/src/components/common/Description.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import React from "react";
+import { DESCRIPTION } from '../../utils/type';
+
+const Description = (props: DESCRIPTION) => {
+  const { bemBlock, tooltip, errorMessage, description, showShortDescription, showLongDescription } = props;
+  return (<>
+    {
+      showShortDescription && tooltip && (
+        <div
+          title='Help Text'
```
Comment (rismehta) [inline packages/react-vanilla-components/src/components/common/Description.tsx:26]: Title should be set via props right ? Also, why is this title not localised value

Comment (amit-mnnit) [reply packages/react-vanilla-components/src/components/common/Description.tsx:26]: Similar to core component https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/fieldTemplates/questionMark.html

Comment (amit-mnnit) [reply packages/react-vanilla-components/src/components/common/Description.tsx]: changed

File: packages/react-vanilla-components/src/components/common/Label.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            {label?.value}
+          </label>
+        )
+      }
+      {
+        description && (
+          <button
+            className={`${bemBlock}__questionmark`}
+            onClick={onClick}
+            aria-label='Toggle Button'
```
[the remark this answers is not in the document]
Comment (amit-mnnit) [reply packages/react-vanilla-components/src/components/common/Label.tsx:37]: changed

**Pull request merged**